frontend/apps/reader/modules/readercoptlistener.lua

```diff
@@ -300,15 +311,29 @@ end
 ReaderCoptListener.onCloseDocument = ReaderCoptListener.unscheduleHeaderRefresh
 ReaderCoptListener.onSuspend = ReaderCoptListener.unscheduleHeaderRefresh
 
+function ReaderCoptListener:addAdditionalHeaderContent(content_func)
+    table.insert(self.external_header_content, content_func)
+end
+
+function ReaderCoptListener:removeAdditionalHeaderContent(content_func)
+    for i, v in pairs(self.external_header_content) do
+        if v == content_func then
+            table.remove(self.external_header_content, i)
```

Comment:
As you made it an array, use `ipairs` instead of `pairs`. (Also when you use the content. Also in footer.)
(Not sure it's best to use an array : if a second plugin use that, depending on ordering, of it one uncheck and check in the menu, the order may change. But we can revisit this storage/insertion later when we need to - ie. using a alpha/numerical key/value: "readtimer" => content_func.)